elide-async/src/test/java/com/yahoo/elide/async/service/AsyncQueryThreadTest.java

```diff
@@ -45,6 +51,53 @@ public void setupMocks() {
 
     @Test
     public void testProcessQueryJsonApi() throws Exception {
+        AsyncQuery queryObj = new AsyncQuery();
+        String responseBody = "{\"data\":"
+                + "[{\"type\":\"book\",\"id\":\"3\",\"attributes\":{\"title\":\"For Whom the Bell Tolls\"}}"
+                + ",{\"type\":\"book\",\"id\":\"2\",\"attributes\":{\"title\":\"Song of Ice and Fire\"}},"
+                + "{\"type\":\"book\",\"id\":\"1\",\"attributes\":{\"title\":\"Ender's Game\"}}]}";
+        ElideResponse response = new ElideResponse(200, responseBody);
+        String query = "/group?sort=commonName&fields%5Bgroup%5D=commonName,description";
+        String id = "edc4a871-dff2-4054-804e-d80075cf827d";
+        queryObj.setId(id);
+        queryObj.setQuery(query);
+        queryObj.setQueryType(QueryType.JSONAPI_V1_0);
+        queryObj.setResultType(ResultType.EMBEDDED);
+
+        when(elide.get(anyString(), any(), any(), anyString(), any())).thenReturn(response);
+        AsyncQueryThread queryThread = new AsyncQueryThread(queryObj, user, elide, runner, asyncQueryDao, "v1",
+                resultStorageEngine);
+        queryResultObj = queryThread.processQuery();
+        assertEquals(queryResultObj.getRecordCount(), 3);
+        assertEquals(queryResultObj.getResponseBody(), responseBody);
+        assertEquals(queryResultObj.getHttpStatus(), 200);
+    }
+
+    @Test
```

Comment:
may be include a comment saying which particular method/scenario is getting tested here.